src/gc-heap-snapshot.cpp

```diff
@@ -106,15 +106,22 @@ struct StringTable {
 };
 
 struct HeapSnapshot {
-    vector<Node> nodes;
-    // edges are stored on each from_node
-
     StringTable names;
```

Comment:
You also might want to consider streaming this field out into the file per-node also, at least in some or all cases, since I think this is often unique? The JSON can probably deal with being a Union{Int,String} field, depending on whether the content was probably unique (or long, like a String) or probably common like a name.

Reply:
Yeah, i think a nice thing here could be to keep some kind of like a bloom filter or something, and in the nodes file, we write out *either* an index into this table, _or_ we write out the string itself.

But thinking more about it, i think that it's maybe even fine to just duplicate the strings over and over. The priorities here are:
A) be fast
B) don't OOM
I don't think that file size reduction is nearly as important as those two.

So if we can intern _some_ of the strings, like you suggested, and write out the rest, that's probably good enough, yeah. Keep common and big strings, and write them out uniquely otherwise.

This is the last major cleanup that I think we should do, otherwise this PR looks good to go.
It seems to be working in its current state in order to avoid OOMs in our production setup.

Reply:
This is addressed in the new PR https://github.com/JuliaLang/julia/pull/52854 by streaming out the string table directly. At the same time, we keep a very limited amount of known strings in memory to reduce duplicates in the string table as much as possible.